src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -420,7 +425,10 @@ export default class FilterOptionsComponent extends Component {
 
     this.config.options[index] = Object.assign({}, this.config.options[index], { selected });
     this.updateListeners();
-    this.setState();
```

Comment:
Yes! Updated this so that setState is only called when using the `storeOnChange` option -- which triggers a new search with the selected options. 

When not using config.storeOnChange, a setState is not needed. The UI changes are now in the onMount (and the selected option is stored in the URL, but that is/was handled outside of the setState)